tests/go/GoErrCheckBearTest.py

```diff
@@ -0,0 +1,109 @@
+from bears.go.GoErrCheckBear import GoErrCheckBear
+from tests.LocalBearTestHelper import verify_local_bear
+
+good_file = """package main
+import "fmt"
+func main() {
+    fmt.Println("Hello, Arch!")
+}
+""".splitlines(keepends=True)
+
+
+bad_file = """package main
+import "os"
+func main() {
+f, _ := os.Open("foo")
+f.Write([]byte("Hello, world."))
+f.Close()
+}
+""".splitlines(keepends=True)
+
+assert_file = """package main
+
+func main() {
+// Type assertions
+  var i interface{}
+  s1 := i.(string)    // ASSERT
+  s1 = i.(string)     // ASSERT
+  s2, _ := i.(string) // ASSERT
+  s2, _ = i.(string)  // ASSERT
+  s3, ok := i.(string)
+  s3, ok = i.(string)
+  switch s4 := i.(type) {
+  case string:
+    _ = s4
+  }
+  _, _, _, _ = s1, s2, s3, ok
+
+}
+""".splitlines(keepends=True)
+
+blank_file = """package main
+import "os"
+func main() {
+
+// Single error return
+  _ = a() // BLANK
+  a()     // UNCHECKED
+
+  // Return another value and an error
+  _, _ = b() // BLANK
+  b()        // UNCHECKED
+
+  // Return a custom error type
+  _ = customError() // BLANK
+  customError()     // UNCHECKED
+  }
+  """.splitlines(keepends=True)
+
+ignorepkg_file = """package main
+
+import (
+  "io"
+  "log"
+  "os"
+)
+
+func main() {
+  f, _ := os.Open("foo")
+  log.Println("opened file")
+  io.Copy(os.Stdout, f)
+}""".splitlines(keepends=True)
+
+GoErrCheckBearTest = verify_local_bear(GoErrCheckBear,
+                                       valid_files=(
+                                           good_file, assert_file,
+                                           blank_file,),
+                                       invalid_files=(bad_file,),
+                                       tempfile_kwargs={"suffix": ".go"})
+
+GoErrCheckBearWithIgnoreTest = verify_local_bear(GoErrCheckBear,
+                                                 valid_files=(good_file,),
+                                                 invalid_files=(bad_file,),
```

Comment:
shouldn't one of the files tested in the previous test go invalid? But none of them are listed here